tests/python_tests/models/test_couplings.py

```diff
@@ -0,0 +1,590 @@
+            if not site1.symmetry == site2.symmetry:
+                continue
+            coupling = couplings.spin_spin_coupling([site1, site2], Jx=Jx, Jy=Jy, Jz=Jz)
+            coupling.test_sanity()
+            if check_dense_blocks:
+                expect = site1.spin_vector.copy()
+                expect[:, :, 0] *= Jx
+                expect[:, :, 1] *= Jy
+                expect[:, :, 2] *= Jz
+                expect = np.tensordot(expect, site2.spin_vector, axes=[2, 2])
+                expect = np.transpose(expect, [0, 2, 3, 1])
+                assert np.allclose(coupling.to_numpy(understood_braiding=True), expect)
+
+    # test correct number of sites
+    with pytest.raises(AssertionError):
+        _ = couplings.spin_spin_coupling([site_list[0]], Jx=1., Jy=1., Jz=1.)
+    with pytest.raises(AssertionError):
+        _ = couplings.spin_spin_coupling([site_list[0]] * 3, Jx=1., Jy=1., Jz=1.)
+
+
+def test_spin_field_coupling(any_backend, np_random):
+    check_dense_blocks = np_random.choice([True, False])
+    site_list = generate_spin_dofs(any_backend)
+    num_sites = min(5, len(site_list))
+    site_list = np_random.choice(site_list, size=num_sites, replace=False)
+    for site in site_list:
+        hx, hy, hz = np_random.random(3)
+        # either SpinSite or SpinHalfFermionSite
+        conserve = site.conserve if isinstance(site, sites.SpinSite) else site.conserve_S
+        if conserve in ['Sz', 'parity']:
+            hx = hy = 0
+        elif conserve in ['SU(2)']:
+            # coupling not allowed
+            continue
+        coupling = couplings.spin_field_coupling([site], hx=hx, hy=hy, hz=hz)
+        coupling.test_sanity()
+        if check_dense_blocks:
+            expect = site.spin_vector
+            expect = hx * expect[:, :, 0] + hy * expect[:, :, 1] + hz * expect[:, :, 2]
+            assert np.allclose(coupling.to_numpy(understood_braiding=True), expect)
+
+    # test correct number of sites
+    with pytest.raises(AssertionError):
+        _ = couplings.spin_field_coupling([site_list[0]] * 2, hx=1.)
+
+
+def test_aklt_coupling(any_backend, np_random):
+    check_dense_blocks = np_random.choice([True, False])
+    site_list = generate_spin_dofs(any_backend)
+    num_sites = min(5, len(site_list))
+    site_list = np_random.choice(site_list, size=num_sites, replace=False)
+    for i, site1 in enumerate(site_list):
+        J = np_random.random()
+        # test different site combinations
+        for site2 in site_list[:i + 1]:
+            if not site1.symmetry == site2.symmetry:
+                continue
+            coupling = couplings.aklt_coupling([site1, site2], J=J)
+            coupling.test_sanity()
+            if check_dense_blocks:
+                expect = np.tensordot(site1.spin_vector, site2.spin_vector, axes=[2, 2])
+                expect = np.transpose(expect, [0, 2, 3, 1])
+                expect += np.tensordot(expect, expect, axes=[[2, 3], [1, 0]]) / 3.
```

Comment:
this is just a copy of the defintion no?
so essentially we just test if ``Coupling.from_dense(h).to_numpy() == h`` (laxly)
That is a useful test and we should do it in the generic Couplings test, but this test doesnt give us more than that, or am I missing something?

To really convince us that we produce the operators we expect we should verify known properties.
- check hermitian
- check known eigenenergies
- check known properties of eigenstates, e.g. in AKLT state total spin of groundstate
- check commutation relations, preferably with SymmetricTensors s

Reply:
these checks might look different depending on the concrete site.
so maybe parametrize over different concrete sites (e.g. spin-1/2, spin-1, spin-5/2, spin-1/2-fermion)
and do case distinctions as needed for the checks listed above

which sites are sensible can of course depend on the coupling, so each test can independently decide which sites to use

basically here in the tests, we dont want to be this general about what the site is, we want to be concrete and verify it works.